### ISSUE TITLE
addition creates recurison error 

### ISSUE THREAD

**Issue by dfee** [sdispater/poetry#379]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: MacOS 10.13
- **Poetry version**: 0.11.4
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**:

```toml
[tool.poetry]
name = "mypkg"
version = "2018.0.0"
description = "mypkg module"
authors = ["Devin Fee"]

[tool.poetry.dependencies]
python = "^3.6.5"
rasa_core = "^0.10.3"

[tool.poetry.dev-dependencies]
pytest = "^3.0"
```

## Issue
Adding `rasa_core` bottoms out the recursion limit:

```
:$ poetry add rasa_core
Using version ^0.10.3 for rasa_core

Updating dependencies
Resolving dependencies... (2.6s)

[RecursionError]
maximum recursion depth exceeded

add [-D|--dev] [--git GIT] [--path PATH] [-E|--extras EXTRAS] [--optional] [--python PYTHON] [--platform PLATFORM] [--allow-prereleases] [--dry-run] [--] <name> (<name>)...
```

My recursion limit is set at 3000, and I updated it to the point where a segfault happens before the recursion limit is hit.

**Comment by dfee** [comment]
Alright, I've narrowed the problem space down quite a bit. The problem appears when keras is version 2.2 (or above):

```toml
[tool.poetry]
name = "demo"
version = "0.1.0"
description = ""
authors = ["Devin Fee"]

[tool.poetry.dependencies]
python = "3.6.5"

keras = "~2.1"
rasa_core = "^0.10.3"

[tool.poetry.dev-dependencies]
pytest = "^3.0"
```

Allowing keras v2.2+ raises a RecursionError, which is effectively what's in rasa_core's setup.py: `"keras~=2.0"` (which appears to mean `keras >= 2.0, < 3.0`).

Strangely, if you were to install every dependency of `rasa_core` *except for* rasa_core itself, then keras can be v2.2+:

```toml
[tool.poetry]
name = "demo"
version = "0.1.0"
description = ""
authors = ["Devin Fee"]

[tool.poetry.dependencies]
python = "^3.6"

jsonpickle = "^0.9.6"
six = "^1.11"
redis = "^2.10"
fakeredis = "^0.10.3"
future = "^0.16.0"
numpy = "^1.15"
typing = "^3.6"
requests = "^2.19"
graphviz = "^0.8.4"
keras = "^2.0" # TODO: revert to: keras = "~2.0"
tensorflow = ">= 1.7, < 1.9"
h5py = "^2.8"
apscheduler = "^3.5"
tqdm = "^4.24"
configargparse = "^0.13.0"
networkx = "^2.1"
fbmessenger = "^5.3"
pykwalify = "<= 1.6.0"
coloredlogs = "^10.0"
"ruamel.yaml" = "^0.15.50"
flask = "^1.0"
flask_cors = "^3.0"
scikit-learn = "^0.19.2"
rasa-nlu = "^0.13.1"
slackclient = "^1.2"
python-telegram-bot = "^10.1"
twilio = "^6.16"
mattermostwrapper = "^2.1"
colorhash = "^1.0"

# rasa_core = "^0.10.3" # TODO: notice how this isn't installed, so keras can be 2.2+

[tool.poetry.dev-dependencies]
pytest = "^3.0"
ipython = "^6.5"
```

**Comment by sdispater** [comment]
This is really strange. I am not sure where the issue is coming from.

I will investigate and get back to you.

**Comment by adithyabsk** [comment]
Do you know when the bug fix with the patch related to this issue will be out? It is blocker to our team adapting poetry

**Comment by sdispater** [comment]
This is fixed in the latest `0.11.5` release :-)
